packages/redux-saga-requests/src/middleware.js

```diff
@@ -111,3 +111,24 @@ export const requestsCacheMiddleware = () => {
     return next(action);
   };
 };
+
+export const serverRequestsFilterMiddleware = ({ serverRequestActions }) => {
+  const actionsToBeIgnored = serverRequestActions.slice();
+
+  return () => next => action => {
+    if (!isRequestAction(action)) {
+      return next(action);
+    }
+
+    const actionToBeIgnoredIndex = actionsToBeIgnored.findIndex(
+      a => a.type === action.type,
+    );
+
+    if (actionToBeIgnoredIndex === -1) {
+      return next(action);
+    }
+
+    actionsToBeIgnored.splice(actionToBeIgnoredIndex, 1);
```

Comment:
What is this? Is this like `unshift?`

Reply:
@megawac sry for late response, I did not notice this. We remove an action from `actionsToBeIgnored` in this middleware not to ignore it again. Why? Imagine you do SSR with dispatched `x` on the server. On the client you want to ignore it but only once! Or the exact number which was run on the server. Next request `x` should not be ignored anymore, for instance imagine someone might want to refresh `x` data.

Reply:
@megawac btw, I am just pushing docs now and new release so you could see the whole ssr algorythm explained if you are curious